### DISCUSSION TITLE
Describe R.nvim relation to LSP and tree-sitter

### DISCUSSION THREAD

**Discussion by TymekDev** [R-nvim/R.nvim#113] [Ideas]
Hi! Knowing Nvim-R was quite robust I was wondering what R.nvim tries to cover. Perhaps a section in a README describing its relation with LSP and tree-sitter would be in order? This way it would be clear(er) what are non-goals of R.nvim (e.g. provide syntax highlighting).

**Comment by jalvesaq** [comment]
The README has a section named _Transitioning from Nvim-R_ listing changes and new features. Mostly, `R.nvim` is simply a conversion from VimScript to Lua, with some of `Nvim-R` features dropped during the conversion. Eventually, I may add new features to `R.nvim`, but I don't plan to do so. My goal when I decided to convert VimScript to Lua was to get help from other people. And it worked! @PMassicotte helped to convert VimScript to Lua, and is very actively fixing bugs and creating new features. He and @wurli added new interesting features using `nvim-treesitter`, @she3o documented many functions, started the reorganization of `rnvimserver` and added some tests, @hongyuanjia wrote some code to keep the plugin working on Windows and other people contributed finding and fixing bugs.

Note that I used to be the maintainer of R runtime files, but syntax highlighting and indenting code wasn't a goal of `Nvim-R` (see https://github.com/R-nvim/R.nvim/issues/98#issuecomment-2017778123).

We don't have any code related to LSP, but the C code in `nvimcom` and `rnvimserver` was adjusted to build strings that represent Lua dictionaries instead of VimScript dictionaries. Hence, if someone in the future wants to convert `nvimcom` + `rnvimserver` into a language server, it will be easier to use [cmp-r](https://github.com/R-nvim/cmp-r) than [cmp-nvim-r](https://github.com/jalvesaq/cmp-nvim-r) as a starting point.

In summary, the `R-nvim` organization is decentralized and doesn't have goals: individual members have. My goal was to avoid the slow death of the Nvim-R project and now my main goal is to help the `R.nvim` project to be bug-free. In my opinion, `R.nvim` is already better than `Nvim-R` but, of course, people using features that were not converted from `Nvim-R` will disagree.

The transition from `Nvim-R` to `R.nvim` is easier for those who already use an `init.lua` to configure Neovim and who are actively trying to replace VimScript plugins with Lua ones. That's why only those with an `init.lua` are seeing the warning that `Nvim-R` was superseded by `R.nvim` (https://github.com/jalvesaq/Nvim-R/blob/master/R/start_nrs.vim#L532).

**Comment by jalvesaq** [comment]
One correction: Nvim-R contributed to Vim/Neovim's default syntax highlighting by highlighting function names.
